gestalt/__init__.py

```diff
@@ -395,15 +395,18 @@ def __get(
             )
         split_keys = key.split(self.__delim_char)
         consider_keys = list()
+        config_val = None
         for split_key in split_keys:
             consider_keys.append(split_key)
             joined_key = self.__delim_char.join(consider_keys)
             config_val = self._get_config_for_key(key=key,
```

Comment:
hmm, does the `self._get_config_for_key` call here not return the `default` value ever? At least thats what i'm inferring from your additional logic on line 406.

Reply:
The problem here is that it returns the default value at the first val, given 'test.foo' and a default, it will never return the value of 'test.foo' always the default.